### ISSUE TITLE
[Bug] await MotionHandle does not complete when calling handle.Cancel()

### ISSUE THREAD

**Issue by AnnulusGames** [AnnulusGames/LitMotion#14]
Await does not complete when MotionHandle is canceled while waiting with await.

```cs
[UnityTest]
public IEnumerator Test_CancelWhileAwait() => UniTask.ToCoroutine(async () =>
{
    var handle = LMotion.Create(0f, 10f, 1f).BindToUnityLogger();
    DelayedCall(0.2f, () => handle.Cancel()).Forget();
    await handle; // never complete
});

async UniTaskVoid DelayedCall(float delay, Action action)
{
    await UniTask.WaitForSeconds(delay);
    action.Invoke();
}
```

**Comment by Akeit0** [comment]
I think creating `bool TreatCancelAsComplete` or `enum CancelBehaviour` and allowing access and modification from the handle would solve the problem.

**Comment by AnnulusGames** [comment]
Fixed in #15

**Comment by Akeit0** [comment]
This doesn't work with WithCancelOnError.
